Auth0/CredentialsManager.swift

```diff
@@ -142,39 +141,40 @@ public struct CredentialsManager {
+    /// - Note: [Auth0 Refresh Tokens Docs](https://auth0.com/docs/tokens/concepts/refresh-tokens)
     #if WEB_AUTH_PLATFORM
-    public func credentials(withScope scope: String? = nil, callback: @escaping (CredentialsManagerError?, Credentials?) -> Void) {
+    public func credentials(withScope scope: String? = nil, minTTL: Int = 0, callback: @escaping (CredentialsManagerError?, Credentials?) -> Void) {
         guard self.hasValid() else { return callback(.noCredentials, nil) }
         if #available(iOS 9.0, macOS 10.15, *), let bioAuth = self.bioAuth {
             guard bioAuth.available else { return callback(.touchFailed(LAError(LAError.touchIDNotAvailable)), nil) }
             bioAuth.validateBiometric {
                 guard $0 == nil else {
                     return callback(.touchFailed($0!), nil)
                 }
-                self.retrieveCredentials(withScope: scope, callback: callback)
+                self.retrieveCredentials(withScope: scope, minTTL: minTTL, callback: callback)
             }
         } else {
-            self.retrieveCredentials(withScope: scope, callback: callback)
+            self.retrieveCredentials(withScope: scope, minTTL: minTTL, callback: callback)
         }
     }
     #else
-    public func credentials(withScope scope: String? = nil, callback: @escaping (CredentialsManagerError?, Credentials?) -> Void) {
+    public func credentials(withScope scope: String? = nil, minTTL: Int = 0, callback: @escaping (CredentialsManagerError?, Credentials?) -> Void) {
         guard self.hasValid() else { return callback(.noCredentials, nil) }
-        self.retrieveCredentials(withScope: scope, callback: callback)
+        self.retrieveCredentials(withScope: scope, minTTL: minTTL, callback: callback)
     }
     #endif
 
-    private func retrieveCredentials(withScope scope: String? = nil, callback: @escaping (CredentialsManagerError?, Credentials?) -> Void) {
-        guard
-            let data = self.storage.data(forKey: self.storeKey),
-            let credentials = NSKeyedUnarchiver.unarchiveObject(with: data) as? Credentials
-            else { return callback(.noCredentials, nil) }
+    private func retrieveCredentials(withScope scope: String?, minTTL: Int, callback: @escaping (CredentialsManagerError?, Credentials?) -> Void) {
+        guard let data = self.storage.data(forKey: self.storeKey),
+            let credentials = NSKeyedUnarchiver.unarchiveObject(with: data) as? Credentials else { return callback(.noCredentials, nil) }
         guard credentials.expiresIn != nil else { return callback(.noCredentials, nil) }
-        guard self.hasExpired(credentials) else { return callback(nil, credentials) }
+        guard self.willExpire(credentials, within: minTTL) ||
+            self.hasExpired(credentials) ||
+            self.hasDifferentScope(credentials, than: scope) else { return callback(nil, credentials) }
```

Comment:
I think `self.hasScopeChanged` is more explicit.

Reply:
Thanks, changed in https://github.com/auth0/Auth0.swift/pull/399/commits/76f4575cf672ad7792eae20d4c64361a0de4a504.